pkg/controller/dataimportcron-controller.go

```diff
@@ -250,18 +251,21 @@ func (r *DataImportCronReconciler) update(ctx context.Context, dataImportCron *c
 
 	importSucceeded := false
 	imports := dataImportCron.Status.CurrentImports
-	if imports != nil {
+	dataVolume := &cdiv1.DataVolume{}
+	dvFound := false
+	if len(imports) > 0 {
 		// Get the currently imported DataVolume
-		dataVolume := &cdiv1.DataVolume{}
 		if err := r.client.Get(ctx, types.NamespacedName{Namespace: dataImportCron.Namespace, Name: imports[0].DataVolumeName}, dataVolume); err != nil {
 			if !k8serrors.IsNotFound(err) {
 				return res, err
 			}
-			log.Info("DataVolume not found for some reason, so let's recreate it", "name", imports[0].DataVolumeName)
-			if err := r.createImportDataVolume(ctx, dataImportCron); err != nil {
-				return res, err
-			}
+			log.Info("DataVolume not found, removing from current imports", "name", imports[0].DataVolumeName)
```

Comment:
So previously DV was recreated, now its entry is removed. Why is that?

Reply:
By user, for example.